File: semantic-api/infrastructure/src/main/java/iscte/ico/semantic/infrastructure/model/Query.java
package iscte.ico.semantic.infrastructure.model;

import org.hibernate.annotations.GenericGenerator;


import javax.persistence.*;
import java.io.Serializable;
import java.util.UUID;

@Entity
@Table(name = "query")
public class Query implements Serializable {

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
            name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator"
    )
    @Column(name = "id", updatable = false, nullable = false)
    private UUID id;
    @Column(nullable = false)
    private String name;
    @Lob
    private Object queryParameters;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Object getQueryParameters() {
        return queryParameters;
    }

    public void setQueryParameters(Object queryParameters) {
        this.queryParameters = queryParameters;
    }
}

File: semantic-api/application/src/main/kotlin/iscte/ico/semantic/application/model/Query.kt
package iscte.ico.semantic.application.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import lombok.Data
import java.util.*

@Data
class Query{

    val id: UUID
    val name: String
    val queryParameters : Array<QueryParameters>

    @JsonCreator
    constructor(
        @JsonProperty("id") id: UUID,
        @JsonProperty("name") name: String,
        @JsonProperty("query_parameters") queryParameters : Array<QueryParameters>
    ){
        this.id = id
        this.name = name
        this.queryParameters = queryParameters
    }
}
